contracts/multisig/src/signing.rs

```diff
@@ -61,14 +65,32 @@ pub fn validate_session_signature(
     signature: &Signature,
     pub_key: &PublicKey,
     block_height: u64,
+    sig_verifier: Option<SignatureVerifier>,
 ) -> Result<(), ContractError> {
     if session.expires_at < block_height {
         return Err(ContractError::SigningSessionClosed {
             session_id: session.id,
         });
     }
 
-    if !signature.verify(&session.msg, pub_key)? {
+    let valid = sig_verifier.map_or_else(
+        || signature.verify(&session.msg, pub_key),
+        |verifier| {
+            verifier
+                .verify_signature(
+                    signature.as_ref().into(),
+                    session.msg.as_ref().into(),
+                    pub_key.as_ref().into(),
+                    signer.to_string(),
+                    session.id,
```

Comment:
Why are `signer_address` and `session_id` relevant?

Reply:
the common prefix people specifically asked for these fields. I think passing more data here is preferable, to make this as generic and flexible as possible. I don't want to leave some stuff out, and then it turns out in the future, some logic needs more information, and then we need to change things and pass more information. I'd rather just pass everything now, and not have to touch this again.

Reply:
Why do they ask for these to verify signatures? I am ok with the idea of sending as much data as possible but still find it weird to include these. If that's the goal, why not send the entire session over?

Reply:
from the SigningSession fields, I think the only one that might make sense to expose is the `chain_name`, I wouldn't like to pass the whole struct as it is because then there is more risk of breaking changes if something changes in the struct

Reply:
the chain name is not necessary, the prover should know the chain name. But if you really want to send it, I'm not opposed. They need the session id to do some internal mapping. They are not actually validating the message, it's a different message actually that's being signed, and I'm pretty sure they need the session id to know what message is being signed. I think the bare minimum to send is session_id, signature and either pubkey or worker address. But again, I'd err on the side of sending everything.